internal/kafka/internal/services/quota/ams_quota_service.go

```diff
@@ -306,3 +340,46 @@ func (q amsQuotaService) DeleteQuota(subscriptionId string) *errors.ServiceError
 	}
 	return nil
 }
+
+func (q amsQuotaService) DeleteQuotaForBillingModel(subscriptionId string, kafkaBillingModel config.KafkaBillingModel) *errors.ServiceError {
+	if subscriptionId == "" {
+		return nil
+	}
+
+	subscription, found, err := q.GetSubscriptionByID(subscriptionId)
+	if err != nil {
+		return errors.GeneralError("failed to delete the quota: %v", err)
+	}
+
+	if !found {
+		return nil
+	}
+
+	if subscription.Plan().ID() == kafkaBillingModel.AMSProduct {
```

Comment:
Is there the possibility that the given provided subscription ID is not the one we want to delete? Is this check needed?

Reply:
Same question applies to the AMS Resource check

Reply:
Hmm, we can probably get rid of this `DeleteQuotaForBillingModel` method.
It was meant to ensure that it didn't delete the subscription if, after a promotion, the `subscription_id` was updated, but the `actual billing model` was still pointing to the old one.

I believe with the latest code it can't happen.

Reply:
:+1: Let's review this

Reply:
Was this reviewed?